src/test/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/processor/IncomingMessageConsumerUnitTest.java

```
@@ -0,0 +1,87 @@
+        ReflectionTestUtils.setField(incomingMessageConsumer, "topicName", "filing-received");
+        ReflectionTestUtils.setField(incomingMessageConsumer, "pollTimeout", 100);
+        incomingMessageConsumer.init();
+
+        verify(consumer).connect();
+
+        ArgumentCaptor<ConsumerConfig> consumerConfigCaptor = ArgumentCaptor.forClass(ConsumerConfig.class);
+        verify(incomingMessageConsumer).createConsumer(consumerConfigCaptor.capture());
+        ConsumerConfig config = consumerConfigCaptor.getValue();
+        assertNotNull(config.getBrokerAddresses());
+        assertEquals(1, config.getBrokerAddresses().length);
+
+        assertEquals(ReflectionTestUtils.getField(incomingMessageConsumer, "groupName"),
+                config.getGroupName());
+        assertEquals(ReflectionTestUtils.getField(incomingMessageConsumer, "brokerAddress"),
+                config.getBrokerAddresses()[0]);
+        assertNotNull(config.getTopics());
+        assertEquals(ReflectionTestUtils.getField(incomingMessageConsumer, "topicName"),
+                config.getTopics().get(0));
+        assertEquals(1, config.getTopics().size());
+        assertEquals(ReflectionTestUtils.getField(incomingMessageConsumer, "pollTimeout"),
+                config.getPollTimeout());
+    }
+
+    @Test
+    void testReadNoMessage() {
+        List<Message> messages = Collections.emptyList();
+        when(consumer.consume()).thenReturn(messages);
+        Collection<ChipsKafkaMessage> result = incomingMessageConsumer.read();
+        assertTrue(result.isEmpty());
+    }
+
+    @Test
+    void testReadValidMessage() {
+        // TODO mock deserializer
+        List<Message> messages = new ArrayList<Message>();
+        when(consumer.consume()).thenReturn(messages);
+        Collection<ChipsKafkaMessage> result = incomingMessageConsumer.read();
+        assertTrue(result.isEmpty());
+    }
```

we should be able to test the error condition as well, the deserialise should throw an exception which we can use to test the catch block in read()

---

The deserialise method is declared private.